src/Sarif.Viewer.VisualStudio/SarifSnapshot.cs

```diff
@@ -0,0 +1,289 @@
+using Microsoft.VisualStudio.TextManager.Interop;
+
+namespace SarifViewer
+{
+    class SarifSnapshot : TableEntriesSnapshotBase
+    {
+        private string _projectName;
+        private readonly List<SarifError> _errors;
+        private Dictionary<string, string> _remappedFilePaths;
+        private List<Tuple<string, string>> _remappedPathPrefixes;
+        private Dictionary<string, NewLineIndex> _fileToNewLineIndexMap;
+
+        internal SarifSnapshot(string filePath, IEnumerable<SarifError> errors)
+        {
+            FilePath = filePath;
+            _errors = new List<SarifError>();
+            _errors.AddRange(errors);
+
+            _remappedFilePaths = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            _remappedPathPrefixes = new List<Tuple<string, string>>();
+
+            _fileToNewLineIndexMap = new Dictionary<string, NewLineIndex>(StringComparer.OrdinalIgnoreCase);
+        }
+
+        public override int Count
+        {
+            get { return _errors.Count; }
+        }
+
+        public IList<SarifError> Errors
+        {
+            get { return _errors; }
+        }
+
+        public string FilePath { get; }
+
+        public override int VersionNumber { get; } = 1;
+
+        public override bool TryGetValue(int index, string columnName, out object content)
+        {
+            content = null;
+
+            if ((index >= 0) && (index < _errors.Count))
+            {
+                if (columnName == StandardTableKeyNames.DocumentName)
+                {
+                    content = FilePath;
+                }
+                else if (columnName == StandardTableKeyNames.ErrorCategory)
+                {
+                    content = Constants.VSIX_NAME;
+                }
+                else if (columnName == StandardTableKeyNames.ErrorSource)
+                {
+                    content = Constants.VSIX_NAME;
+                }
+                else if (columnName == StandardTableKeyNames.Line)
+                {
+                    content = _errors[index].LineNumber;
+                }
+                else if (columnName == StandardTableKeyNames.Column)
+                {
+                    content = _errors[index].ColumnNumber;
+                }
+                else if (columnName == StandardTableKeyNames.Text)
+                {
+                    content = _errors[index].Message;
+                }
+                else if (columnName == StandardTableKeyNames.ErrorSeverity)
+                {
+                    content = _errors[index].IsError ? __VSERRORCATEGORY.EC_ERROR : __VSERRORCATEGORY.EC_WARNING;
+                }
+                else if (columnName == StandardTableKeyNames.Priority)
+                {
+                    content = _errors[index].IsError ? vsTaskPriority.vsTaskPriorityHigh : vsTaskPriority.vsTaskPriorityMedium;
+                }
+                else if (columnName == StandardTableKeyNames.ErrorSource)
```

Comment:
Woah! You already treated `ErrorSource` above (line 71), and you set `content` to `Constants.VSIX_NAME`.